src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -19,15 +22,18 @@
     private final BlobServiceClient storageClient;
     private final BlobProcessor blobProcessor;
     private final BlobReadinessChecker blobReadinessChecker;
+    private final EventRecordRepository eventRecordRepository;
 
     public ContainerProcessor(
         BlobServiceClient storageClient,
         BlobProcessor blobProcessor,
-        BlobReadinessChecker blobReadinessChecker
+        BlobReadinessChecker blobReadinessChecker,
+        EventRecordRepository eventRecordRepository
```

Comment:
Most jobs/processors/services use `EnvelopeService` instead of using repositories directly.
Shall we do this here as well?

Reply:
ok. at least it will be consistent